### PR TITLE
IGNITE-20262 Refuse accepting partition Raft commands when not enough schemas are available

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-20262

Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/InterceptingActionRequestProcessor.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * Extension of the standard {@link ActionRequestProcessor} that allows to add some interception logic.
+ *
+ * @see ActionRequestInterceptor
+ */
+public class InterceptingActionRequestProcessor extends ActionRequestProcessor {
+    private final ActionRequestInterceptor interceptor;
+
+    /**
+     * Constructor.
+     */
```
Comment (tkalkirill) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/InterceptingActionRequestProcessor.java:36]:
```suggestion
    /** Constructor. */
```

**Review by tkalkirill — APPROVED** [review]

File: modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/ActionRequestInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.apache.ignite.raft.jraft.rpc.RaftServerService;
+import org.apache.ignite.raft.jraft.rpc.RpcContext;import org.apache.ignite.raft.jraft.rpc.RpcRequestClosure;
+import org.apache.ignite.raft.jraft.rpc.RpcRequests.AppendEntriesRequest;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Intercepts {@link ActionRequest}s as they come in. It might be used to handle such a request in a non-standard
+ * way (like returning EBUSY under special circumstances instead of the standard behavior).
+ */
+public interface ActionRequestInterceptor {
```
Comment (alievmirza) [inline modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/ActionRequestInterceptor.java:30]: Let's mark this and `AppendEntriesRequestInterceptor` with `@SuppressWarnings("InterfaceMayBeAnnotatedFunctional")`

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/schema/CheckCatalogVersionOnActionRequest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                            "Metadata not yet available, group '%s', required level %d; rejecting ActionRequest with EBUSY.",
+                            request.groupId(), requiredCatalogVersion
+                    );
+            }
+        }
+
+        return null;
+    }
+
+    private boolean isMetadataAvailableFor(int requiredCatalogVersion) {
```
Comment (alievmirza) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/schema/CheckCatalogVersionOnActionRequest.java:77]: This code duplicates `org.apache.ignite.internal.table.distributed.schema.CheckCatalogVersionOnAppendEntries#isMetadataAvailableFor`, lets put it in a common place
